riscv/src/large_field/bootloader.rs

```diff
@@ -523,15 +544,26 @@ shutdown_start:
 "#,
     );
 
-    // Go over all registers except the PC
-    let register_iter = REGISTER_NAMES.iter().take(REGISTER_NAMES.len() - 1);
+    // Go over all memory registers
+    for (i, reg) in REGISTER_MEMORY_NAMES.iter().enumerate() {
+        let reg = reg.strip_prefix("main::").unwrap();
+        bootloader.push_str(&format!(
+            "assert_bootloader_input 0, {}, 1, {};\n",
+            Register::from(reg).addr(),
```

Comment:
can you check and leave a TODO on this `Register::from` implementation to maybe use the `REGISTER*NAMES`? It would be less likely to break in the future should one of them change without the other